### ISSUE TITLE
Leveraging bootstrap's features to join the filter clear button with the filter text input

### ISSUE THREAD

**Issue by MrCsabaToth** [vedmack/yadcf#134]
I'm using some text filters and unfortunately the UI experience is different for all major browser platforms (FF, Chrome, IE): the button isn't vertically aligned with the corresponding input, the "x" is not in center (slightly weaving to different direction on different browsers). I don't know how much the frameworks I use cause this, but I do use bootstrap 3.1+. Taking advantage of bootstrap's styles, the clear button can nicely join the filter input without any sorcery.

I'm far from developing a complete pull request, because I only patched my particular case: text input filter. Such a development maybe could be a part of a bootstrap support in general. But my change is easy to explain: one liner (two liner actually).

So right now in appendFilters (Line 1280) adds an input element for text filters on line 1768:
https://github.com/vedmack/yadcf/blob/master/lab/jquery.dataTables.yadcf.js#L1768

$(filter_selector_string).find(".yadcf-filter").after("<input value=\"" + filter_reset_button_text + "\" type=\"button\" " + " id=\"yadcf-filter-" + table_selector_jq_friendly + "-" + column_number + "-reset\"" +
								"onclick=\"yadcf.stopPropagation(event);yadcf.textKeyUP('" + table_selector_jq_friendly + "', event, 'clear'); return false;\" class=\"yadcf-filter-reset-button\">");

The trick is to wrap the input element around with a span class=input-group-btn, and make the input a button like this:
						    $(filter_selector_string).find(".yadcf-filter").after("<span class=\"input-group-btn\"><button value=\"" +  + "\" type=\"button\" " + " id=\"yadcf-filter-" + table_selector_jq_friendly + "-" + column_number + "-reset\"" +
								"onclick=\"yadcf.stopPropagation(event);yadcf.textKeyUP('" + table_selector_jq_friendly + "', event, 'clear'); return false;\" class=\"yadcf-filter-reset-button btn btn-xs btn-default\">" + filter_reset_button_text + "</button></span>");

I wish I have time for introducing bootstrap specific features for each filter, and then could be a generic options which would turn these on. But try this small modification how nice it can look. It adds tiny rounded corners and stuff.

**Comment by MrCsabaToth** [comment]
So the problem is that originally the HTML structure is like this:
<div id="yadcf-filter-wrapper--reportTable-6" class="yadcf-filter-wrapper">
    <input id="yadcf-filter--reportTable-6" class="yadcf-filter" type="text" .../>
    <input id="yadcf-filter--reportTable-6-reset" class="yadcf-filter-reset-button" type="button".../>
</div>

For bootstrap an extra wrapper (the input-group-bt span) would be needed just around the reset button:
<div id="yadcf-filter-wrapper--reportTable-6" class="yadcf-filter-wrapper">
    <input id="yadcf-filter--reportTable-6" class="yadcf-filter" type="text" .../>
    <span class="input-group-btn">
        <input id="yadcf-filter--reportTable-7-reset" class="yadcf-filter-reset-button btn btn-xs btn-default" type="button".../>
    </span>
</div>

**Comment by vedmack** [comment]
Someone opened a possibly similar issue (https://github.com/vedmack/yadcf/issues/139) with the wrap around the reset button (and changing from input to button) and he said that wrapping is not a must, please contact him or try to think about alternative solution to wrapping because I prefer not to add the wrap because it might break the ui for other users and personally I think it can be designed well even without the wrap. If you think that the wrap is a must and there is no other solution you can reopen the issue.

**Comment by MrCsabaToth** [comment]
pinin says: "wrap is not required.
It is only for bootstrap input group."

which in my reading means that it is required for bootstrap input groups, but not essential for anything else.
http://www.tutorialspoint.com/bootstrap/bootstrap_input_groups.htm
See "Button addons" section which I followed so far.

The good news is that I tried and if I give "input-group" class to the wrapper div, and "input-group-addon" class to the reset button (instead of an extra wrapper span), it works given that I adjust the padding. So I've found a solution which seems to work. But that's not how originally bootstrap meant to be.
